src/hotspot/share/oops/method.cpp

```diff
@@ -1317,12 +1323,18 @@ void Method::set_code(const methodHandle& mh, CompiledMethod *code) {
   OrderAccess::storestore();
   mh->_from_compiled_entry = code->verified_entry_point();
   OrderAccess::storestore();
-  // Instantly compiled code can execute.
-  if (!mh->is_method_handle_intrinsic())
-    mh->_from_interpreted_entry = mh->get_i2c_entry();
+
   if (mh->is_continuation_enter_intrinsic()) {
-    // this is the entry used when we're in interpreter-only mode; see InterpreterMacroAssembler::jump_from_interpreted
+    assert(mh->_from_interpreted_entry == NULL, "initialized incorrectly"); // see link_method
+
+    // This is the entry used when we're in interpreter-only mode; see InterpreterMacroAssembler::jump_from_interpreted
     mh->_i2i_entry = mh->get_i2c_entry();
+    OrderAccess::storestore();
+    // This must come last, as it is what's tested in LinkResolver::resolve_static_call
+    mh->_from_interpreted_entry = mh->get_i2c_entry();
+    OrderAccess::storestore();
```

Comment:
Technically this needs a release-store to pair with the load-acquire.

But what subsequent store is this final store being ordered with? If there is none then it would suffice for this sequence to be just:
```
    mh->_i2i_entry = mh->get_i2c_entry();
    // This must come last, as it is what's tested in LinkResolver::resolve_static_call
    Atomic::release_store(&mh->_from_interpreted_entry , mh->get_i2c_entry());
```

Reply:
Done.